db.go

```diff
@@ -136,6 +136,11 @@ func (db *DB) replayFunction() func(Entry, valuePointer) error {
 		} else {
 			nv = vp.Encode()
 			meta = meta | bitValuePointer
+			// Update vhead. If the crash happens while replay was in progess
+			// and the head is not updated, we will end up replaying all the
+			// files again.
+			y.AssertTrue(!vp.Less(db.vhead))
+			db.vhead = vp
```

Comment:
We could use the `db.UpdateHead` function here but using that function requires locking `db.vhead`.
https://github.com/dgraph-io/badger/blob/158d9271e1052082d6377ffa56483650129b726c/db.go#L674-L691

I could move the lock outside of the function but this would cause contention on the lock. The db.Lock would be held while we're iteration over all the pointers. I prefer to keep it as it is.